### PR TITLE
[IMP] spreadsheet: replace pivot style with table style

### PR DESCRIPTION
**Pull request by hokolomopo**
Task: [3709540](https://www.odoo.com/web#id=3709540&cids=1&menu_id=4720&action=333&active_id=2328&model=project.task&view_type=form)
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by LucasLefevre — APPROVED** [review]
waiting the next o-spreadsheet release to be merged

File: addons/spreadsheet/static/tests/utils/pivot.js [hunk 1]
```diff
@@ -86,3 +86,38 @@ export async function createSpreadsheetWithPivot(params = {}) {
     await waitForDataLoaded(model);
     return { model, env, pivotId };
 }
+
+/**
+ * Return the zone that contains all the cells of the given data source
+ *
+ * TODO: Un-duplicate this once this or #50623 is merged
```
Comment (pro-odoo) [inline addons/spreadsheet/static/tests/utils/pivot.js:93]: It's merged

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js [hunk 2]
```diff
@@ -243,7 +244,21 @@ export class PivotCorePlugin extends OdooCorePlugin {
...
         }
 
-        this._addBorders(sheetId, position, table);
+        const pivotZone = {
+            top: position.row,
+            bottom: position.row + pivotCells[0].length - 1,
+            left: position.col,
+            right: position.col + pivotCells.length - 1,
+        };
+        const numberOfHeaders = table.getNumberOfHeaderRows() - 1; // -1 to not include measures in the table headers
```
Comment (pro-odoo) [inline addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:253]: `const numberOfHeaders = table.getColHeaders().length;`

**Review by pro-odoo — APPROVED** [review]

File: addons/spreadsheet/static/src/helpers/constants.js [hunk 3]
```diff
@@ -5,9 +5,16 @@ import { _t } from "@web/core/l10n/translation";
 
 export const DEFAULT_LINES_NUMBER = 20;
 
-export const HEADER_STYLE = { fillColor: "#E6F2F3" };
-export const TOP_LEVEL_STYLE = { bold: true, fillColor: "#E6F2F3" };
```
Comment (rrahir) [inline addons/spreadsheet/static/src/helpers/constants.js:9]: this variable is still used in list_core_plugin, it should not be removed.

**Review by rrahir — CHANGES_REQUESTED** [review]

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js (hunk 2 above)
Comment (hokolomopo) [reply addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:253]: `table.getColHeaders().length`and `table.getNumberOfHeaderRows()` give the exact same result, I'm not sure what's teh difference ?

File: addons/spreadsheet/static/src/helpers/constants.js (hunk 3 above)
Comment (hokolomopo) [reply addons/spreadsheet/static/src/helpers/constants.js:9]: we probably should also change list to be Tables now that I think about it

File: addons/spreadsheet/static/src/helpers/constants.js [hunk 4]
```diff
@@ -5,9 +5,18 @@ import { _t } from "@web/core/l10n/translation";
 
 export const DEFAULT_LINES_NUMBER = 20;
 
-export const HEADER_STYLE = { fillColor: "#E6F2F3" };
 export const TOP_LEVEL_STYLE = { bold: true, fillColor: "#E6F2F3" };
-export const MEASURE_STYLE = { fillColor: "#E6F2F3", textColor: "#756f6f" };
+
+export const PIVOT_TABLE_CONFIG = {
+    hasFilters: false,
+    totalRow: false,
```
Comment (rrahir) [inline addons/spreadsheet/static/src/helpers/constants.js:12]:
```suggestion
    totalRow: true,
```

File: addons/spreadsheet/static/src/helpers/constants.js [hunk 5]
```diff
@@ -5,9 +5,18 @@ import { _t } from "@web/core/l10n/translation";
...
+    firstColumn: true,
+    lastColumn: false,
```
Comment (rrahir) [inline addons/spreadsheet/static/src/helpers/constants.js:14]:
```suggestion
    lastColumn: true,
```

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js [hunk 6]
```diff
@@ -243,7 +244,21 @@ export class PivotCorePlugin extends OdooCorePlugin {
...
         }
 
-        this._addBorders(sheetId, position, table);
+        const pivotZone = {
+            top: position.row,
+            bottom: position.row + pivotCells[0].length - 1,
+            left: position.col,
+            right: position.col + pivotCells.length - 1,
+        };
+        const numberOfHeaders = table.getColHeaders().length - 1; // -1 to not include measures in the table headers
```
Comment (rrahir) [inline addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:253]:
```suggestion
        const numberOfHeaders = table.getColHeaders().length;
```
We looked at it withvincent, we think the meaasure should be part of the header (as it is a pivot header from a formula POV)

**Review by rrahir — CHANGES_REQUESTED** [review]

File: addons/spreadsheet/static/src/helpers/constants.js (hunk 3 above)
Comment (rrahir) [reply addons/spreadsheet/static/src/helpers/constants.js:9]: I think we can leave it for later, not a 100% sure of what result we want... We should discuss this with @Francois-frgi

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js (hunk 6 above)
Comment (LucasLefevre) [reply addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:253]: we're going to change our mind every 2 weeks ? :smile:  (see task pad and chatter)

It's technically part of the headers yes, but it's way less (not at all) important that they stand out as the dimensions does. It's the same measure repeated, over and over again. The color and bold etc. catches the eyes but it really shouldn't.

It's certainly not at the same level of importance as the group headers.

**Comment by rrahir** [comment]
@robodoo r+

**Review by rrahir — APPROVED** [review]

**Pull request closed without merging**